airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -252,16 +260,49 @@ def create_api_key_authenticator(model: ApiKeyAuthenticatorModel, config: Config
                 parameters=model.parameters,
             )
         )
-        return ApiKeyAuthenticator(api_token=model.api_token, request_option=request_option, config=config, parameters=model.parameters)
+        return ApiKeyAuthenticator(
+            token_provider=token_provider
+            if token_provider is not None
+            else InterpolatedStringTokenProvider(api_token=model.api_token, config=config, parameters=model.parameters),
+            request_option=request_option,
+            config=config,
+            parameters=model.parameters,
+        )
+
+    def create_session_token_authenticator(
+        self, model: SessionTokenAuthenticatorModel, config: Config, **kwargs
+    ) -> Union[ApiKeyAuthenticator, BearerAuthenticator]:
+        login_requester = self._create_component_from_model(model=model.login_requester, config=config, name="login_requester")
+        token_provider = SessionTokenProvider(
+            login_requester=login_requester,
+            session_token_path=model.session_token_path,
+            expiration_time=parse_duration(model.expiration_time) if model.expiration_time else None,
+            parameters=model.parameters,
+            message_repository=self._message_repository,
+        )
+        if model.request_authentication.type == "Bearer":
+            return ModelToComponentFactory.create_bearer_authenticator(
+                BearerAuthenticatorModel(type="BearerAuthenticator", api_token=""), config, token_provider=token_provider
+            )
+        else:
+            return ModelToComponentFactory.create_api_key_authenticator(
+                ApiKeyAuthenticatorModel(type="ApiKeyAuthenticator", api_token="", inject_into=model.request_authentication.inject_into),
+                config=config,
+                token_provider=token_provider,
+            )
 
     @staticmethod
     def create_basic_http_authenticator(model: BasicHttpAuthenticatorModel, config: Config, **kwargs) -> BasicHttpAuthenticator:
         return BasicHttpAuthenticator(password=model.password, username=model.username, config=config, parameters=model.parameters)
 
     @staticmethod
-    def create_bearer_authenticator(model: BearerAuthenticatorModel, config: Config, **kwargs) -> BearerAuthenticator:
+    def create_bearer_authenticator(
+        model: BearerAuthenticatorModel, config: Config, token_provider: Optional[TokenProvider] = None, **kwargs
```

Comment:
I think we should update the interface to pass in a `Union[SessionTokenAuthenticatorModel, TokenProvider]`